cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2013,6 +2030,47 @@ if(ARROW_WITH_BZ2)
   include_directories(SYSTEM "${BZIP2_INCLUDE_DIR}")
 endif()
 
+macro(build_utf8proc)
+  message(STATUS "Building utf8proc from source")
+  set(UTF8PROC_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/utf8proc_ep-install")
+  if(MSVC)
+    set(UTF8PROC_STATIC_LIB "${UTF8PROC_PREFIX}/lib/utf8proc_static.lib")
+  else()
+    set(
+      UTF8PROC_STATIC_LIB
+      "${UTF8PROC_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}utf8proc${CMAKE_STATIC_LIBRARY_SUFFIX}"
+      )
+  endif()
+
+  set(UTF8PROC_CMAKE_ARGS
+      ${EP_COMMON_TOOLCHAIN}
+      "-DCMAKE_INSTALL_PREFIX=${UTF8PROC_PREFIX}"
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
```

Comment:
Could you use `lib` if we use `lib` in `UTF8PROC_STATIC_LIB`?
Or could you use `${CMAKE_INSTALL_LIBDIR}` in `UTF8PROC_STATIC_LIB`?